### PR TITLE
Open API Updates AAH-57

### PR DESCRIPTION
**Pull request by alikins**
WIP for getting a openapi spec we can use with c.r.c

Issue: AAH-57

### CODE REVIEW

File: galaxy_ng/app/settings.py [hunk 1]
```diff
@@ -82,3 +82,20 @@
 # ----------------------
 X_PULP_CONTENT_HOST = "localhost"
 X_PULP_CONTENT_PORT = 24816
+
+# works
+# SPECTACULAR_SETTINGS__TITLE = "Automation Hub API __TITLE"
```
Comment (awcrosby) [inline galaxy_ng/app/settings.py:87]: are these lines something that can be removed since title is below?

File: galaxy_ng/app/settings.py [hunk 2]
```diff
@@ -82,3 +82,20 @@
...
+
+SPECTACULAR_SETTINGS = {
+    "TITLE": "Automation Hub API",
+    "DESCRIPTION": "Fetch, Upload, Organize, and Distribute Ansible Collections",
+    "VERSION": "v3",
+    "LICENSE": {
+        "name": "GPLv2+",
+        "url": "https://raw.githubusercontent.com/pulp/pulpcore/master/LICENSE",
```
Comment (awcrosby) [inline galaxy_ng/app/settings.py:95]: should we link to galaxy_ng license?

File: galaxy_ng/app/common/openapi.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+def preprocess_debug_logger(endpoints, **kwargs):
+    """Return an iterable of (path, path_regex, method, callback) tuples"""
+
+    log.debug("kwargs: %s", repr(kwargs))
+
+    return [
+        (path, path_regex, method, callback) for path, path_regex, method, callback in endpoints
+        if not log.debug('path=%s, path_regex=%s, methos=%s, callback=%s',
+                         path, path_regex, method, callback)
```
Comment (awcrosby) [inline galaxy_ng/app/common/openapi.py:15]: curious what `preprocess_debug_logger()` is used for... and does `log.debug(...)` sometimes evaluate to a boolean false by not returning any result?

**Review by awcrosby — APPROVED** [review]
nothing blocking merging from my pov, but did add a couple clarifying questions.

File: galaxy_ng/app/settings.py (hunk 1 above)
Comment (alikins) [reply galaxy_ng/app/settings.py:87]: Yeah, that is meant as an example of how to set drf_spectular settings vis env variables / dynaconf. I'll rephrase to make that clearer.

File: galaxy_ng/app/settings.py (hunk 2 above)
Comment (alikins) [reply galaxy_ng/app/settings.py:95]: Ah, yeah. Good catch.

File: galaxy_ng/app/common/openapi.py (hunk 3 above)
Comment (alikins) [reply galaxy_ng/app/common/openapi.py:15]: nah, log.debug() always returns nothing. That is leftover from some c&p from the other preprocess function

Comment (alikins) [reply galaxy_ng/app/common/openapi.py:15]: I take it back, forgot it was in the 'if' clause of the generator, and the if `not log.debug()` ends up if True, so all the elements are returned.

Comment (alikins) [reply galaxy_ng/app/common/openapi.py:15]: I've refactored that a bit to look less weird. I'll update after a './compose build' cycle to catch up some deps to test.

**Comment by hendersonreed** [comment]
A lot of the api's endpoints take specific parameters when POSTing- is there a way to generate request samples for those? Currently it looks like there's only request samples for the PATCH and PUT listings.

**Comment by alikins** [comment]
@hendersonreed Yeah, lots of stuff is incomplete at the moment. It can be corrected, either by tweaks to the spec generator, or direct markup on the viewsets and serializers.

**Comment by hendersonreed** [comment]
Seems to me that ideally we do those QoL improvements before merging this? I'm happy to contribute code if I can, since I'll also be a consumer of the final result.

**Pull request merged**